### ISSUE TITLE
Add to style guide: "Use italics for something you see and bold for something you click"

### ISSUE THREAD

**Issue by seanmadsen** [civicrm/civicrm-dev-docs#174]
In #173 @stevekessler [said the following](https://github.com/civicrm/civicrm-dev-docs/issues/173#issuecomment-305989740):

> In my own training and support writing "style" I use italics. I think using italics for something you see and bold for something you click would be very clear. Right now the guide calls for bold for something you click. A collapsed fieldset would still be ambiguous because it is sort of both.

Since the Style Guide should (IMHO) require a bit more vetting than your run-of-the-mill docs page, I've cherry-picked this specific suggestion into its own ticket so that it's easier for us to discuss as a group. 

@stevekessler would you care to suggest some specific language/changes/placement for how we'd add this concept to the Style Guide. Seems like our changes should focus on [this section](https://docs.civicrm.org/dev/en/latest/best-practices/documentation-style-guide/#describing-the-civicrm-user-interface).

I'd like to get a go-ahead from other docs people like @michaelmcandrew @joannechester (and anyone else who wants to weigh in) before changing the Style Guide. 
In #173 @stevekessler [said the following](https://github.com/civicrm/civicrm-dev-docs/issues/173#issuecomment-305989740):

> In my own training and support writing "style" I use italics. I think using italics for something you see and bold for something you click would be very clear. Right now the guide calls for bold for something you click. A collapsed fieldset would still be ambiguous because it is sort of both.

Since the Style Guide should (IMHO) require a bit more vetting than your run-of-the-mill docs page, I've cherry-picked this specific suggestion into its own ticket so that it's easier for us to discuss as a group. 

@stevekessler would you care to suggest some specific language/changes/placement for how we'd add this concept to the Style Guide. Seems like our changes should focus on [this section](https://docs.civicrm.org/dev/en/latest/best-practices/documentation-style-guide/#describing-the-civicrm-user-interface).

I'd like to get a go-ahead from other docs people like @michaelmcandrew @joannechester (and anyone else who wants to weigh in) before changing the Style Guide.

**Comment by seanmadsen** [comment]
I support this change. 

@stevekessler I think the way you laid it conceptually out is clear and actually not ambiguous. Everything you click you also see. So if the docs are discussing a step where you click, then it's bold.

**Comment by PalanteJon** [comment]
+1 for this. I try to follow the style good when posting on SE; I was thinking about this same issue myself recently as a result. 
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by michaelmcandrew** [comment]
we used to use italics to refer to other parts of the book when our documentation platform did not support hyperlinks (strange but true). Since we can now use links to refer to other parts of the book, it frees up italics, so i would say go for it.

**Comment by joannechester** [comment]
I just want to clarify what is meant here by using using what I think is proposed to restyle the [Widgets](https://docs.civicrm.org/user/en/latest/contributions/widgets/) section from the book :

> To create a widget go to the configuration for your Contribution Page. From the tabs at the top of the page select **Widgets.** Tick _Enable Widget_ to get started.
> 
> Enter a _Title_ for your widget. It will generally be the same name as your Contribution Page but it does not have to be.
> 
> You can enter the _URL to Logo Image_. To use this feature you will need to have an uploaded an image to your web server and then obtained the URL (web address) for the image. You may need to ask your system administrator for help getting this information.
> 
> Enter a _Button Title_ that will be used for a potential donor to click to go to your donate page. In the above example, the button title is Contribute.
> 
> You are given an editor to enter as much _About_ text as you want. You will want to adjust how much text you have based on where you want to use your Widget.
> 
> When you are ready to preview your Widget or when you are finished and want to update the code for the Widget click **Save and Preview**.
> 
> When selecting the code use the **Select Code** link to make sure you have all of the necessary code.
> 
> At the bottom of the configuration page, you can expand the **Edit Widget Colors** fieldset.
> 
> The fields in this fieldset provide a way to modify the colors for each area of the Widget you are creating. You will need to get hexadecimal (or hex) codes for each color.  [W3schools](https://www.w3schools.com/colors/colors_picker.asp) provides a color picker that will give you the hex values for colors you select.
> 
> A designer that you have worked with on your website may have specific standards for the colors that should be used.
> 
> Click Save and Preview when you are ready to see how your changes look.
> 
> ## Using widgets with WordPress
> 
> WordPress has a plugin that is installed in WordPress [CiviCRM Contribution Page Widget](https://wordpress.org/plugins/civicrm-contribution-page-widget/) that lets you embed a Widget from CiviCRM anywhere at a WorderPress "widget" can go.

I started out thinking I would agree with the suggestion, but having tried it out I don't. I don't think italics stand out enough from normal text.

When I read something like the style for the first time, visually interesting text helps to keep me focussed.  Also, if I was setting up a widget and just wanted to check up on something then I might be searching for information on a particular field (although the in page help icons do away with this need to a certain extent) 

I think **bold** for one and  _**bold and italics**_ for the other would be a better solution.  (Not sure which way around, perhaps **_bold and italics_** invites action and so should be for things you click on.)

Also, as a BTW, is there a reason that the external links are styled differently in the section. One is italicised and the other isn't.

**Comment by stevekessler** [comment]
@joannechester  For your last comment, you caught me in an inconsistency.  

I like bold and italics for things you click on.  The problem as I see it is that italics do not currently have a set meeting in the guide. Bold does. We would need to find all the bold and then make them bold and italic or the meanings will be mixed in the guide.  

@seanmadsen can we automate finding the current bold text and making it bold and italic or something like that?  

-Steve

**Comment by seanmadsen** [comment]
### Thoughts on programmatically altering text formatting

> can we automate finding the current bold text and making it bold and italic

I like the way you're thinking @stevekessler! This would be a bit of a challenge, but possible, theoretically...

First of all, there's the challenge of ignoring bold text which does not actually represent a description of the CiviCRM interface, but is just bold because someone wanted to emphasize something. Short of using some natural language processing, I think we'd need to live with these cases being converted to bold-italic, which I think is probably fine. 

Now, on to the challenge of actually find bold text...

Here's a somewhat primitive stab at finding bold text with regular expressions

```
grep -PIro '\*\*(?!\*)(\w((?!\*\*).)*)\*\*(?!\*)' .
```

It could be tweaked a bit to better deal with edge cases, but so far it finds 2317 cases. Wow! That tells me that if we were to do this, we'd need to be pretty confident in the conversion process because manually reviewing every single change would probably not be feasible. 

Regular expressions are a somewhat blunt tool for a job like this though. If we want more confidence in the replacements we'd really want to use a tool which actually *parses* the mardown. 

[Pandoc scripting](http://pandoc.org/scripting.html) seems like it would be a great tool for this job, but someone would need to learn how to use it, including learning/knowing Haskell or Python. It could be a fun project though!

In summary, I think this is possible, but would be a bit of a project.

**Comment by stevekessler** [comment]
@seanmadsen I think that if bold is being used in the "wrong" ways then it will get caught in this and it will need to be cleaned up.  

I do not know if this is any good but I did find this for Pandoc with PHP (https://github.com/vinai/pandocfilters-php).  I looked to see if my little bit Python knowledge would but enough but I do not think so.  

Thanks,
Steve

**Comment by stevekessler** [comment]
I found the bulleted list here (https://docs.civicrm.org/user/en/latest/the-user-interface/searching/#quick-search) to be a possible exception for the bold.  Should _Peter_ in each of these places be bold?  I think that the bold is very clear and works but I am not sure how this fits with my style guide.
